contracts/express/AxelarValuedExpressExecutable.sol

```diff
@@ -203,31 +204,37 @@ abstract contract AxelarValuedExpressExecutable is ExpressExecutorTracker, IAxel
         _executeWithToken(sourceChain, sourceAddress, payload, symbol, amount);
     }
 
-    function _transferToExecutor(
+    function _allocateValue(
+        string calldata sourceChain,
+        string calldata sourceAddress,
+        bytes calldata payload
+    ) internal virtual {}
+
+    function _transferFromExecutor(
         address expressExecutor,
         address tokenAddress,
         uint256 value
-    ) internal {
+    ) internal virtual {
         if (value == 0) return;
 
         if (tokenAddress == address(0)) {
-            payable(expressExecutor).safeNativeTransfer(value);
+            if (msg.value < value) revert InsufficientValue();
```

Comment:
Should the value be exact?

Reply:
What if there is more value provided to make another GMP call and pay gas? I think we should allow for extra value to be passed.